src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3058,26 +3060,32 @@ ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> rec
                 int addedCount = 0;
                 foreach (ParameterSymbol param in recordParameters)
                 {
-                    var property = new SynthesizedRecordPropertySymbol(this, param, diagnostics);
-                    _ = memberSignatures.TryGetValue(property, out var existingMember);
-                    existingMember ??= getInheritedMember(property, this);
+                    bool isInherited = false;
+                    var syntax = param.GetNonNullSyntaxNode();
+                    var property = new SynthesizedRecordPropertySymbol(this, syntax, param, isOverride: false, diagnostics);
```

Comment:
>SynthesizedRecordPropertySymbol [](start = 39, length = 31)

Consider using SignatureOnlyPropertySymbol to check for existing member. I am planning to switch all checks like that in this function to use SignatureOnly symbols. I think this will eliminate some overhead and will add some predictability in terms of what is happening and when.

Reply:
That seems like a separable change.

---
In reply to: [446540258](https://github.com/dotnet/roslyn/pull/45336#discussion_r446540258) [](ancestors = 446540258)